blockchain/vmQueryGetter.go

```diff
@@ -0,0 +1,170 @@
+package blockchain
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-sdk-go/builders"
+	"github.com/multiversx/mx-sdk-go/data"
+)
+
+const (
+	okCodeAfterExecution = "ok"
+	internalError        = "internal error"
+	vmQuerylogPath       = "blockchain/vmQueryGetter"
+)
+
+type vmQueryGetter struct {
+	proxy Proxy
+	log   logger.Logger
+}
+
+// NewVmQueryGetter creates a new instance of the vmQueryGetter type
+func NewVmQueryGetter(proxy Proxy) (*vmQueryGetter, error) {
+	if check.IfNil(proxy) {
+		return nil, ErrNilProxy
+	}
+
+	client := vmQueryGetter{
+		proxy: proxy,
+	}
+	client.log = logger.GetOrCreate(vmQuerylogPath)
```

Comment:
get the logger as param?

Reply:
we should provide the logger as a param. Valid on all SDK use cases.
Reason: when implementing parallel system tests execution, we want to store all log lines and print them grouped when the tests end.

Reply:
done